.github/actions/commit_and_push/commit_and_push.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+# Inputs passed as environment variables:
+# GITHUB_TOKEN, BRANCH_NAME, COMMIT_MESSAGE, ADD_FILES
+
+# Stage and detect changes
+git add "$ADD_FILES"
+CHANGED=$(git diff --cached --name-only)
+if [ -z "$CHANGED" ]; then
+  echo "No changes to commit"
+  exit 0
+fi
+
+# Get expectedHeadOid using local git (faster, no API calls)
+EXPECTED_HEAD=$(git rev-parse HEAD)
+echo "Using local HEAD SHA: $EXPECTED_HEAD"
+
+# Build GraphQL payload in temp file
+tmp=$(mktemp)
+cat <<EOF > "$tmp"
+{"query":"mutation(\$input:CreateCommitOnBranchInput!){createCommitOnBranch(input:\$input){commit{oid url}}}","variables":{"input":{"branch":{"repositoryNameWithOwner":"$GITHUB_REPOSITORY","branchName":"$BRANCH_NAME"},"fileChanges":{"additions":[
```

Comment:
Is it possible to format this nicely with proper GraphQL indentation? That would make this part much more declarative

Reply:
done